File: src/client/src/types/blocks/blockView.ts
import { Aspect } from "../common/aspect";
import { AttributeTypeReferenceView } from "../common/attributeTypeReferenceView";
import { ImfType } from "../common/imfType";
import { RdlClassifier } from "../common/rdlClassifier";
import { RdlPurpose } from "../common/rdlPurpose";
import { EngineeringSymbol } from "./engineeringSymbol";
import { TerminalTypeReferenceView } from "./terminalTypeReferenceView";

export interface BlockView extends ImfType {
  classifiers: RdlClassifier[];
  purpose: RdlPurpose | null;
  notation: string | null;
  symbol: EngineeringSymbol | null;
  aspect: Aspect | null;
  terminals: TerminalTypeReferenceView[];
  attributes: AttributeTypeReferenceView[];
  kind: string;
}

File: src/server/Tyle.Api/Blocks/BlockView.cs
using Tyle.Api.Common;
using Tyle.Core.Blocks;
using Tyle.Core.Common;

namespace Tyle.Api.Blocks;

public class BlockView : ImfType
{
    public ICollection<RdlClassifier> Classifiers { get; set; } = new List<RdlClassifier>();
    public RdlPurpose? Purpose { get; set; }
    public string? Notation { get; set; }
    public EngineeringSymbol? Symbol { get; set; }
    public Aspect? Aspect { get; set; }
    public ICollection<TerminalTypeReferenceView> Terminals { get; set; } = new List<TerminalTypeReferenceView>();
    public ICollection<AttributeTypeReferenceView> Attributes { get; set; } = new List<AttributeTypeReferenceView>();
    public string Kind => nameof(BlockView);
}